docs/AccessEnvironmentVariables.md

```diff
@@ -0,0 +1,174 @@
+
+# ✅ Good: Using Datadog.get_environment_variable
+api_key = Datadog.get_environment_variable('DD_API_KEY')
+timeout = Datadog.get_environment_variable('DD_TIMEOUT', '30')
+```
+
+This is enforced by the `CustomCops::EnvUsageCop` cop.
+
+### 2. Supported Configurations Registry
+
+All environment variables that start with `DD_` or `OTEL_` must be registered in the `supported-configurations.json` file to be accessible through `Datadog.get_environment_variable`.
+
+### 3. Automatic Code Enforcement
+
+Custom RuboCop cops automatically detect and prevent direct `ENV` usage:
+
+- `CustomCops::EnvUsageCop` - Prevents direct `ENV` access and auto-corrects to use `Datadog.get_environment_variable`
+- `CustomCops::EnvStringValidationCop` - Validates that environment variable strings are registered in supported configurations
+
+## How It Works
+
+The configuration inversion system works as follows:
+
+1. **Environment Variable Request**: Code calls `Datadog.get_environment_variable('DD_SOME_VAR')`
+2. **Validation**: The `ConfigHelper` checks if the variable is in the supported configurations
+3. **Access Control**:
+   - If the variable starts with `DD_` or `OTEL_` but is NOT in supported configurations → returns `nil` (or raises error in test environment)
+   - If the variable is supported or doesn't start with `DD_`/`OTEL_` and is not an alias → returns the value
+4. **Alias Resolution**: Checks for any configured aliases if the primary variable is not set
+5. **Deprecation Logging**: Logs deprecation warnings for deprecated variables
+
+## Adding New Environment Variables
+
+To add support for a new environment variable:
+
+### Step 1: Add to supported-configurations.json
+
+Edit the `supported-configurations.json` file and add your variable:
+
+```json
+{
+  "supportedConfigurations": {
+    "DD_YOUR_NEW_VARIABLE": {
+      "version": ["A"]
+    }
+  }
+}
+```
+
+#### Configuration Structure
+
+- **supportedConfigurations**: Maps variable names to configuration metadata
+  - `version`: (Currently always set to `["A"]`) Array indicating which tracer versions change the behavior of the configuration
+
+  In the future, the structure will also contain more information such as the type, the default value...
+
+- **aliases**: Maps canonical variable names to arrays of alias names
+  ```json
+  "aliases": {
+    "DD_SERVICE": ["OTEL_SERVICE_NAME"]
+  }
+  ```
+
+- **deprecations**: Adds a log message to deprecated environment variables.
+  ```json
+  "deprecations": {
+    "DD_OLD_VARIABLE": "Please use DD_NEW_VARIABLE",
+    "DD_REMOVED_VARIABLE": "This feature will be removed in the next release"
+  }
+  ```
+
+### Step 2: Generate Configuration Assets
+
+Run the rake task to generate the Ruby configuration assets:
+
+```bash
+bundle exec rake local_config_map:generate
+```
+
+This task generates `lib/datadog/core/configuration/supported_configurations.rb` ahead of time, so the tracer does not need to parse the JSON file every time.
```

Comment:
This is super awesome! We can definitely use this pattern more to reduce the startup overhead, and improve runtime performance in some cases.